docs/resources/server/damagetypes.md

```diff
@@ -0,0 +1,127 @@
+  // "death.examplemod.example" (with swapped-out mod ids and names).
+  "message_id": "example",
+  // Whether this damage type's damage amount scales with difficulty or not. Valid values are:
+  // - "never": The damage value remains the same on any difficulty. Common for player-caused damage types.
+  // - "when_caused_by_living_non_player": The damage value is scaled if the entity is caused by a
+  //   living entity of some sort, including indirectly (e.g. an arrow shot by a skeleton), that is not a player.
+  // - "always": The damage value is always scaled. Commonly used by explosion-like damage.
+  "scaling": "when_caused_by_living_non_player",
+  // The amount of exhaustion caused by receiving this kind of damage.
+  "exhaustion": 0.1,
+  // The damage effects (currently only sound effects) that are applied when receiving this kind of damage. Optional.
+  // Valid values are "hurt" (default), "thorns", "drowning", "burning", "poking", and "freezing".
+  "effects": "hurt",
+  // The death message type. Determines how the death message is built. Optional.
+  // Valid values are "default" (default), "fall_variants", and "intentional_game_design".
+  "death_message_type": "default"
+}
+```
+
+:::tip
+The `scaling`, `effects` and `death_message_type` fields are internally controlled by the enums `DamageScaling`, `DamageEffects` and `DeathMessageType`, respectively. These enums can be [extended][extenum] to add custom values if needed.
+:::
+
+The same format is also used for vanilla's damage types, and pack developers can change these values if needed.
+ 
+## Creating and Using Damage Sources
+
+`DamageSource`s are usually created on the fly when `Entity#hurt` is called. Be aware that since damage types are a [datapack registry][dr], you will need a `RegistryAccess` to query them, which can be obtained via `Level#registryAccess`. To create a `DamageSource`, call the `DamageSource` constructor with up to four parameters:
+
+```java
+DamageSource damageSource = new DamageSource(
+        // The damage type holder to use. Query from the registry. This is the only required parameter.
+        registryAccess.registryOrThrow(Registries.DAMAGE_TYPE).getHolderOrThrow(EXAMPLE_DAMAGE),
+        // The entity causing the damage. This isn't always applicable (e.g. when falling out of the world)
+        // and may therefore be null. Optional, defaults to null.
+        null,
+        // The direct entity. For example, if a skeleton shot you, the skeleton would be the causing entity
+        // (= the parameter above), and the arrow would be the direct entity (= this parameter). Similar to
+        // the causing entity, this isn't always applicable and therefore nullable. Optional, defaults to null.
+        null,
+        // The damage source position. This is rarely used, one example would be intentional game design
+        // (= nether beds exploding). Nullable and optional, defaulting to null.
+        null
+);
```

Comment:
You should, preferably, be getting the `DamageSource` through `DamageSources` and not throwing on the registry each time. This is bad practice.

Probably should add a NeoForge method to take in all three values.

Reply:
Resolved via Discord discussion (second-to-last paragraph of linked message): https://discord.com/channels/313125603924639766/1185208532014997524/1256350727656439841